blocks/gu_spdetails/block_gu_spdetails.php

```diff
@@ -302,7 +302,7 @@ public static function return_dates($modname, $activity) {
                                    $activity->overridecutoffdate : $activity->cutoffdate;
                 $date->gradingduedate = (!empty($activity->gradingduedate)) ?
                                         $activity->gradingduedate :
-                                        (!(empty($date->duedate)) ? $date->duedate + (86400 * 14) : '0');
+                                        (!(empty($date->duedate)) ? $date->duedate : '0');
```

Comment:
For assignment, `$date->gradingduedate` should only be based on `$activity->gradingduedate`. If this is empty, just set it to 0.

`$date->gradingduedate = (!empty($activity->gradingduedate)) ? $activity->gradingduedate : '0';`

Reply:
Fixed at commit [1b50080](https://github.com/uofg-andf/andfmoodle/pull/184/commits/1b50080a3cab7bf33148cba444d1b1d2f7579d88)